tools/stats/sccache_stats_to_benchmark_format.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from typing import Any, Dict
+import glob
+import json
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+
+def flatten_data(d: Dict[str, Any]) -> Dict[str, Any]:
+    # Flatten the sccache stats data from a possibly nested dictionary to a flat
+    # dictionary.  For example, the input:
+    # {
+    #     "cache": {
+    #         "hit": 1,
+    #         "miss": 2,
+    #     },
+    # }
+    # will be transformed to:
+    # {
+    #     "cache_hit": 1,
+    #     "cache_miss": 2,
+    # }
+    flat_data = {}
+    for key, value in d.items():
+        if isinstance(value, dict):
+            for k, v in flatten_data(value).items():
+                flat_data[f"{key}_{k}"] = v
+        else:
+            flat_data[key] = value
+    return flat_data
+
+
+def main() -> None:
+    records = []
+    for file in glob.glob(str(REPO_ROOT / "sccache-stats-*.json")):
+        with open(file) as f:
+            data = json.load(f)
+
+            # I don't know what sccache info will be most useful yet, and the
+            # sccache json has a decent number of keys, so just flatten the data
+            # and store all of it
+            records.append(
+                {
+                    "benchmark": {
+                        "name": "sccache_stats",
+                    },
+                    "metric": {
+                        "name": "sccache_stats",
```

Comment:
Is there total build time already in the sccache stats?  I think that metric is important enough to be include here, i.e.

```
{
    "benchmark": {
        "name": "sccache_stats",
    },
    "metric": {
        "name": "Build time in minutes",
        "benchmark_values": [20], <-- This is an array of values here
        "extra_info": flatten_data(data),
    },
}
```

Reply:
I don't think so, here's an example I'm looking at
https://gha-artifacts.s3.amazonaws.com/pytorch/pytorch/11860249614/1/artifact/sccache-stats-linux-focal-cuda12.1-py3.10-gcc9-sm75-33055085149.json

Maybe compile requests would work here?  I assume that I'd have to get the build time from something else

Reply:
We could get the build time from the job table I think, but it could be complicated because you will need to go to specific steps to get that info, so it makes sense to gather it here.

Reply:
I'm going to leave as is for now, since we don't actually have anything that currently measures CI build time in place.  An acceptable proxy for now is the time of the build job or the build step